include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
+                        for(uint32_t ampIndex = 0; ampIndex < Amplitude::numComponents; ++ampIndex)
+                        {
+                            std::string dir = dataLabels(ampIndex);
+                            mesh_amp[dir].setUnitSI(factor);
+                            mesh_amp[dir].setPosition(std::vector<double>{0.0, 0.0, 0.0});
+                            ::openPMD::Dataset dataset_amp = ::openPMD::Dataset(datatype_amp, extent_amp);
+                            mesh_amp[dir].resetDataset(dataset_amp);
 
-                    for(uint32_t detectorDim = 0; detectorDim < 3; ++detectorDim)
-                    {
-                        splash::Dimensions offset(detectorDim, 0, 0);
-                        splash::Selection dataSelection(
-                            bufferSizeDetector,
-                            componentSizeDetector,
-                            offset,
-                            strideDetector);
-
-                        hdf5DataFile.write(
-                            currentStep,
-                            radSplashType,
-                            3,
-                            dataSelection,
-                            (meshesPathName + dataLabelsDetectorDirection(detectorDim)).c_str(),
-                            detectorPositions.data());
-
-                        /* save SI unit as attribute together with data set */
-                        const picongpu::float_64 factorDirection = 1.0;
-                        hdf5DataFile.writeAttribute(
-                            currentStep,
-                            radSplashType,
-                            (meshesPathName + dataLabelsDetectorDirection(detectorDim)).c_str(),
-                            "unitSI",
-                            &factorDirection);
-
-                        /* position */
-                        std::vector<float_X> positionMesh(simDim, 0.0); /* there is no offset - zero */
-                        hdf5DataFile.writeAttribute(
-                            currentStep,
-                            splashFloatXType,
-                            (meshesPathName + dataLabelsDetectorDirection(detectorDim)).c_str(),
-                            "position",
-                            1u,
-                            splash::Dimensions(simDim, 0, 0),
-                            &(*positionMesh.begin()));
+                            // select data
+                            for(int copyIndex = 0; copyIndex < N_tmpBuffer; ++copyIndex)
+                            {
+                                tmpBuffer[copyIndex]
+                                    = ((picongpu::float_64*)
+                                           values.data())[ampIndex + Amplitude::numComponents * copyIndex];
+                            }
+                            // write actual data
+                            mesh_amp[dir].storeChunk(::openPMD::shareRaw(tmpBuffer), offset_amp, extent_amp);
```

Comment:
```suggestion
                            mesh_amp[dir].storeChunk(tmpBuffer, offset_amp, extent_amp);
```
`storeChunk` accepts vectors, no need to call `shareRaw` explicitly
How large are those datasets? You could use the Span-based `storeChunk` API which gives the backend the chance to allocate and handle write buffers. That avoids data duplication in ADIOS2. If that's desirable, I can write that part.

Reply:
Regarding the Span-based storeChunk API:
The data can vary in size `= 6 * N_omege * N_n` doubles from  `N_omege` between 128 till 2048 and `N_n` between 1 and 1024 (currently). Yes, If there is a better option, could you provide a link to an example implementation (or as you offered just commit a pull request to this ranch 😃)

Reply:
I will prepare a PR. This stuff is a bit tricky since you want to make sure that the vector is used as a fallback if the efficient implementation is not needed, and also you need to use some template magic to maintain backward compatibility with openPMD 0.12.0
Example for this is [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/6f6af676586284ef8b2c17107a9c00a957e2f906/include/picongpu/plugins/openPMD/writer/ParticleAttribute.hpp#L120)

Reply:
Thanks!

Reply:
https://github.com/PrometheusPi/picongpu/pull/5